### DISCUSSION TITLE
Make `sea-orm-cli` DB backends optional

### DISCUSSION THREAD

**Discussion by Expurple** [SeaQL/sea-orm#2225] [Ideas]
**EDIT**: resolved in #2367

## The issue

My project uses a local `sea-orm-cli` (#1889) and only connects to PostgreSQL. `Cargo.toml` has the following:

```toml
[dependencies]
sea-orm = { version = "0.12.15", features = [
  "debug-print",
  "macros",
  "postgres-array",
  "runtime-tokio-rustls",
  "sqlx-postgres",
] }
sea-orm-cli = { version = "=0.12.15", default-features = false, features = [
  "codegen",
  "cli",
  "runtime-tokio-rustls",
] }
sea-orm-migration = { version = "0.12.15", features = [
    "sqlx-postgres",
    "runtime-tokio-rustls",
    "with-chrono",
] }
```

If I comment out `sea-orm-cli`, then `sea-orm` and `sea-orm-migration` behave nicely and only pull in `sqlx-postgres`. However, `sea-orm-cli` pulls in `sqlx-sqlite` and `sqlx-mysql`. I'd like to avoid that.

## Details

Unlike other SeaQL crates, `sea-orm-cli` doesn't have any feature flags to control this behavior. The problem seems to originate here, where it pulls in `sea-schema/sqlx-all`:

```toml
# sea-orm-cli/Cargo.toml
[features]
codegen = ["sea-schema/sqlx-all", "sea-orm-codegen"]
```

Why does it need that? Can this be avoided?

With some guidance, it's possible that I can contribute the missing flags myself.

**Comment by Expurple** [comment]
Reported again by another user: #2320

**Comment by xamgore** [comment]
```diff
-sea-orm = { version = "0.12.15", features = [
+sea-orm = { version = "=1.0.0", default-features = false, features = [
```

**Comment by Expurple** [comment]
Resolved in https://github.com/SeaQL/sea-orm/pull/2367
